java-squid/src/main/java/org/sonar/java/se/SymbolicValue.java

```diff
@@ -128,10 +124,8 @@ public void computedFrom(List<SymbolicValue> symbolicValues) {
       if (data == null || !data.equals(booleanConstraint)) {
 
         // store constraint only if symbolic value can be reached by a symbol.
-        if (programState.values.containsValue(this)) {
-          Map<SymbolicValue, Object> temp = Maps.newHashMap(programState.constraints);
-          temp.put(this, booleanConstraint);
-          return ImmutableList.of(new ProgramState(programState.values, temp, programState.visitedPoints, programState.stack));
+        if (programState.isReachable(this)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1066)